plugins/kotlin/uast/uast-kotlin-fir/test/org/jetbrains/kotlin/idea/fir/uast/FirUastApiTest.kt

```diff
@@ -33,6 +33,146 @@ open class FirUastApiTest : AbstractFirUastTest() {
+        fun testNameContainingFile() {
+            doCheck("uast-kotlin/testData/NameContainingFile.kt", ::checkCallbackForNameContainingFile)
+        }
+
+        @TestMetadata("DefaultImpls.kt")
+        fun testDefaultImpls() {
+            doCheck("uast-kotlin/testData/DefaultImpls.kt", ::checkCallbackForDefaultImpls)
+        }
+
+        @TestMetadata("ParameterPropertyWithAnnotation.kt")
+        fun testParameterPropertyWithAnnotation() {
+            doCheck("uast-kotlin/testData/ParameterPropertyWithAnnotation.kt", ::checkCallbackForParameterPropertyWithAnnotation)
+        }
+
+        @TestMetadata("TypeInAnnotation.kt")
+        fun testTypeInAnnotation() {
+            doCheck("uast-kotlin/testData/TypeInAnnotation.kt", ::checkCallbackForTypeInAnnotation)
+        }
+
+        @TestMetadata("ElvisType.kt")
+        fun testElvisType() {
+            doCheck("uast-kotlin/testData/ElvisType.kt", ::checkCallbackForElvisType)
+        }
+
+        @TestMetadata("IfStatement.kt")
+        fun testIfStatement() {
+            doCheck("uast-kotlin/testData/IfStatement.kt", ::checkCallbackForIfStatement)
+        }
+
+        @TestMetadata("WhenStringLiteral.kt")
+        fun testWhenStringLiteral() {
+            doCheck("uast-kotlin/testData/WhenStringLiteral.kt", ::checkCallbackForWhenStringLiteral)
+        }
+
+        @TestMetadata("WhenAndDestructing.kt")
+        fun testWhenAndDestructing() {
+            doCheck("uast-kotlin/testData/WhenAndDestructing.kt", ::checkCallbackForWhenAndDestructing)
+        }
+
+        @TestMetadata("BrokenMethod.kt")
+        fun testBrokenMethod() {
+            doCheck("uast-kotlin/testData/BrokenMethod.kt", ::checkCallbackForBrokenMethod)
+        }
+
+        @TestMetadata("EnumValuesConstructors.kt")
+        fun testEnumValuesConstructors() {
+            doCheck("uast-kotlin/testData/EnumValuesConstructors.kt", ::checkCallbackForEnumValuesConstructors)
+        }
+
+        @TestMetadata("EnumValueMembers.kt")
+        fun testEnumValueMembers() {
+            doCheck("uast-kotlin/testData/EnumValueMembers.kt", ::checkCallbackForEnumValueMembers)
+        }
+
+        @TestMetadata("SimpleAnnotated.kt")
+        fun testSimpleAnnotated() {
+            doCheck("uast-kotlin/testData/SimpleAnnotated.kt", ::checkCallbackForSimpleAnnotated)
+        }
+
+        @TestMetadata("SuperCalls.kt")
+        fun testSuperCalls() {
+            doCheck("uast-kotlin/testData/SuperCalls.kt", ::checkCallbackForSuperCalls)
+        }
+
+        @TestMetadata("Anonymous.kt")
+        fun testAnonymous() {
+            doCheck("uast-kotlin/testData/Anonymous.kt", ::checkCallbackForAnonymous)
+        }
+
+        @TestMetadata("TypeAliases.kt")
+        fun testTypeAliases() {
+            doCheck("uast-kotlin/testData/TypeAliases.kt", ::checkCallbackForTypeAliases)
+        }
+
+// TODO: vararg, arrayOf call inside annotation
+//        @TestMetadata("AnnotationComplex.kt")
```

Comment:
What do you think about storing `// FIR_IGNORE` directive inside testdata file instead of directly commenting out each test?

Reply:
We've tried that directive before, and it turned out that such addition will alter the render, in particular, hash value inside local variables, which depends on line number or text location. So, I'd rather avoid changing (seemingly) unrelated things.

But, this reminds me of where to put the logic to handle failures, thanks.

Reply:
Hm, will adding `// FIR_IGNORE` to the end of the tesdata file will break the line numbers/hashes? The same approach is used in find usages tests where line numbers are also important

Reply:
Oh...... bottom of the testdata file... 🤦  Yeah, that should work.

Well, I put a whitelist in FIR UAST, along with short investigation notes. I can do the same, like putting some notes after `// FIR_IGNORE`. I can use that later. :)

One thing I'm worried is, the same testdata file is used by a number of test suites, which may use the same directive-based filtering. Adding that directive may affect all of those tests?

Reply:
> Adding that directive may affect all of those tests?
it may. In that case, let's leave as it is now.